knowledge_graph/classifier/keyword.py

```diff
@@ -73,10 +103,10 @@ def split_by_case_handling(labels: list[str]) -> tuple[list[str], list[str]]:
                         ord(char) > 127 for char in label
                     ):
                         # Labels including uppercase or non-ASCII characters are added to the case-sensitive list
-                        case_sensitive_labels.append(re.escape(label))
+                        case_sensitive_labels.append(make_separator_flexible(label))
```

Comment:
now, making the labels separator-flexible is a side effect of splitting by case handling.could it be better instead to first make all labels separator-flexible below, and then split them, without this hidden side-effect? (or split them and then make them separator-flexible, I guess)